### DISCUSSION TITLE
HMS-1600-4T not connecting / "TX SPI timeout"

### DISCUSSION THREAD

**Discussion by Lineflyer** [tbnobody/OpenDTU#1471] [Q&A] [answered]
After using openDTU for a longer time with HM-series (NRF-module), I changed my setup.
I bought a "ready-to-go" openDTU (with CMT-module + external antenna) to connect a HMS-1600-4T.

I failed to get the rectifier connected. I doublechecked the serial number and all settings and found no obvious error.
The rectifier LED is flashing green and its producing, distance is 40 meter without any obstacle.

Looking into the console I am wondering about the "TX SPI timeout" at identical timestamp as the "TX System ConfigPara"/"TX RealTimeRunData". Is this indicating a (hardware) problem?

```
16:01:55.511 > Fetch inverter: <intentionally removed>
16:01:55.511 > Request SystemConfigPara
16:01:55.648 > TX ChannelChangeCommand 868.00 MHz --> 56 91 62 32 03 80 10 28 79 02 15 21 20 14 57 
16:01:55.648 > TX SPI Timeout
16:01:55.717 > RX Period End
16:01:55.717 > All missing
16:01:55.717 > Nothing received, resend count exeeded
16:01:55.850 > TX RealTimeRunData 868.00 MHz --> 15 91 62 32 03 80 10 28 79 80 0B 00 65 45 6D D4 00 00 00 00 00 00 00 00 0A 54 5A 
16:01:55.850 > TX SPI Timeout
16:01:56.061 > RX Period End
16:01:56.061 > All missing
16:01:56.061 > Nothing received, resend whole request
16:01:56.061 > TX RealTimeRunData 868.00 MHz --> 15 91 62 32 03 80 10 28 79 80 0B 00 65 45 6D D4 00 00 00 00 00 00 00 00 0A 54 5A 
16:01:56.061 > TX SPI Timeout
16:01:56.584 > RX Period End
16:01:56.584 > All missing
16:01:56.584 > Nothing received, resend whole request
```

**Comment by Lineflyer** [comment]
Irgendeine Idee? Gerne auch auf deutsch.
Deutet "TX SPI timeout" auf ein Hardware-Problem hin? Das CMT-Modul wird als konfiguriert und verbunden angezeigt.

**Reply by stefan123t** [reply]
Hast Du den GPIO3 vom CMT2300A an einen in der pin_mapping.json konfigurierten Pin des ESP32 verbunden / geprüft?
Es gibt / gab ältere Platinen bei denen diese Verbindung (noch) nicht vorgesehen war.
Muss aber sein sonst bekommt die OpenDTU nicht mit wenn der CMT ein Paket versendet / empfangen hat. Wird ähnlich genutzt wie der IRQ beim NRF.

**Reply by Lineflyer** [reply, marked as answer]
Ich habe die DTU komplett vorgefertigt gekauft.
Ich habe inzwischen eine andere vom selben Lieferanten und dort tritt der Fehler (bei gleichem PIN-Mapping) nicht auf. Ich denke, hier ist es wirklich ein HW-Defekt.
